### ISSUE TITLE
Support "local transport"?

### ISSUE THREAD

**Issue by ceball** [dailymuse/git-fit#2]
Hi

Short story: how would you feel about supporting 'local transport' in git-fit?

Long story...

I've been using git-media to store large ("media") files in git. However, as your README points out, the smudge/clean filters must operate frequently, which is bad news if you have many media files. Your README also points out that "smudge/clean filters have a sometimes unintuitive execution model"; for me, the problem is that after a ``git media sync``, something like ``git update-index --assume-unchanged -- (media files)`` is run (e.g. https://github.com/alebedev/git-media/blob/master/lib/git-media/sync.rb#L52), which I think makes it easy to mistakenly delete your data (e.g. via ``git stash``). A further problem I have with the smudge/clean filters is that they don't seem to operate consistently across different versions of git, although I did not analyze this carefully, and presumably this problem has gone away now (but at least with git 1.8.4 on windows, the filters did not run for all necessary operations).

I like the idea of git-fit (avoid smudge/clean filters, store metadata in a simple json file), but I don't use s3, so to try out git-fit I had to add support for 'local transport'. However, as I've never seen go code before, I just hacked in the support so that I could test if git-fit would be suitable for me in other regards. I may be able to clean that up and make a pull request, but it would require quite a lot of my time, and I would also likely need some help (i.e. code review and correction...). Therefore, before starting that, I'm writing to see how you view support for 'local transport' (i.e. storing data files on a local disk rather than s3) in the first place?

To me, local transport seems useful in conjunction with e.g. peer-to-peer file synchronization (e.g. Syncthing or bittorrent sync). git-fit would copy files (i.e. files named with the hash of their contents) to a local directory (rather than s3), and that directory would be synchronized automatically by the file synchronization tool. The synchronization would be very fast for machines on the same network, and also would not rely on an external service like s3. If the local directory were on a filesystem allowing insertion only (no editing), data would be safe (or otherwise git-fit's local transport could be careful about never overwriting a file).

**Comment by ysimonson** [comment]
Yes, a PR for local transports would be great! I'm happy to review the code.

For the most part, a local transport simply needs to implement the `Transport` interface. A little refactoring would be needed to allow user customization of what transport they're interested in as well.
